content/dev/api-design-elao-team.md

```diff
@@ -0,0 +1,110 @@
+
+## Normalizers/Serializers : composant `Serializer` de Symfony ou JMSSerializer ?
+
+__Maxime S.__ : Aujourd'hui je n'utilise plus que le composant `Serializer` de Symfony, mais seulement comme un outil technique. J'entends par là que je ne mets plus aucune logique métier dans mes _normalizers_. Si je dois sérializer un objet complexe (comme des données agrégées par exemple), c'est un service dédié qui sera chargé de le construire et c'est cet objet que je passe ensuite directement au _Serializer_. Je n'écris quasiment plus de _normalizers_. Pour retourner des données après un appel `GET` (_queries_), je m'appuie sur des _converters_ qui créent la ressource à partir de la requête, laquelle est transmise au _Serializer_ de Symfony pour la retourner au format JSON. Pour les appels en écriture (_commands_), j'hydrate/désérialize au moyen du _Serializer_  Symfony un DTO de payload à partir de la requête, passé ensuite au validateur de Symfony, puis injecté dans une _command_ pour être traité par le _handler_ approprié. Noter que la désérialization doit être permissive, car je confie la validation du payload aux validateurs de Symfony et on s'autorise donc à instancier un payload invalide. Pour cela, je tire parti d'une mise à jour incluse dans Symfony 3.4 qui permet de passer outre la vérification du typage lors de la sérialization (Cf. [PR 8515](https://github.com/symfony/symfony-docs/pull/8515)).
+
+## Communication développeurs Backend/Frontend : des conseils ?
+
+__Yves__ : J'ai plutôt tendance à privilégier la discussion orale plutôt qu'une documentation "anémique" à outrance. Mais je dois avouer que j'ai essentiellement travaillé dans des petites équipes où la communication n'était pas entravée.
+
+## Et GraphQL dans tout ça ?
+
+__Yves__ : C'est une philosophie différente de REST, qui mérite que l'on s'y intéresse. Mais j'attends également avec impatience les apports du protocole HTTP/2 et je souhaite notamment voir s'il permettra de limiter les appels HTTP, qui est une des problématiques qu'entend adresser GraphQL.
+
+__Maxime S.__ : Hormis les articles de François Zaninotto que j'ai cités, je n'ai pas encore pris le temps de me pencher sérieusement sur la question. GraphQL laisse entrevoir de très belles promesses, mais je m'interroge notamment sur la sécurisation des API et les performances, en raison des requêtes potentiellement lourdes que peut induire GraphQL.
```

Comment:
Ordre des questions : Maxime n'a pas encore cité les articles de Zaninotto à ce moment-là ;) >> to fix !